Wikipedia/Code/ArticleViewController+Editing.swift

```diff
@@ -57,26 +57,34 @@ extension ArticleViewController {
         let navigationController = WMFComponentNavigationController(rootViewController: editVC, modalPresentationStyle: .overFullScreen)
         navigationController.view.isOpaque = false
         navigationController.view.backgroundColor = .clear
-       let needsIntro = !UserDefaults.standard.wmf_didShowTitleDescriptionEditingIntro()
-       if needsIntro {
-           navigationController.view.alpha = 0
-       }
-        let showIntro: (() -> Void)? = {
+        let needsIntro = !UserDefaults.standard.wmf_didShowTitleDescriptionEditingIntro()
+        if needsIntro {
+            navigationController.view.alpha = 0
+        }
+
+        let showIntro: (@escaping () -> Void) -> Void = { completion in
             let welcomeVC = DescriptionWelcomeInitialViewController.wmf_viewControllerFromDescriptionWelcomeStoryboard()
             welcomeVC.completionBlock = {
+                completion()
             }
             welcomeVC.apply(theme: self.theme)
             navigationController.present(welcomeVC, animated: true) {
                 UserDefaults.standard.wmf_setDidShowTitleDescriptionEditingIntro(true)
                 navigationController.view.alpha = 1
             }
         }
+
         present(navigationController, animated: !needsIntro) {
             if needsIntro {
-                showIntro?()
+                showIntro {
+                    editVC.showTempAccountToast()
```

Comment:
I notice this toast doesn't appear if you tap "skip" in the welcomeVC.

I think this would be more reliable if you triggered the toast instead from `viewDidAppear` within `DescriptionEditViewController`. That way you don't have to adjust the presentation logic here outside of the description editor view (for the toast at least).